data/src/test/kotlin/com/antosik/benchproject/data/favourite/repo/FavouriteMoviesDataRepositoryTest.kt

```diff
@@ -0,0 +1,53 @@
+package com.antosik.benchproject.data.favourite.repo
+
+import com.antosik.benchproject.data.movies.common.database.dao.MovieDao
+import com.antosik.benchproject.data.popular.movies.entity.database.MovieEntity
+import com.antosik.benchproject.data.popular.movies.entity.database.toDomain
+import com.antosik.benchproject.domain.popular.movies.entity.MovieModel
+import io.mockk.coEvery
+import io.mockk.every
+import io.mockk.mockk
+import io.mockk.mockkStatic
+import io.mockk.unmockkAll
+import kotlinx.coroutines.flow.first
+import kotlinx.coroutines.flow.flowOf
+import kotlinx.coroutines.test.runTest
+import org.amshove.kluent.shouldBeEqualTo
+import org.junit.jupiter.api.AfterEach
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.DynamicTest
+import org.junit.jupiter.api.TestFactory
+
+internal class FavouriteMoviesDataRepositoryTest {
+
+    val dao: MovieDao = mockk()
+    val tested = FavouriteMoviesDataRepository(dao)
+
+    @BeforeEach
+    fun beforeEach() {
+        mockkStatic(List<MovieEntity>::toDomain)
+    }
+
+    @AfterEach
+    fun afterEach() {
+        unmockkAll()
+    }
+
+    @TestFactory
+    fun `SHOULD return expected favorite movies WHEN specific data is present in the database`() =
```

Comment:
do we really need a dynamic test here ? do we have any logic that is different when the list is empty or not ?

Reply:
To be honest we didn't have any logic when the list is empty or not. I would like just to test both cases but if you think one test will be enough I can go with it.

Reply:
yes but in my opinion it is useless, as long as we don't have any logic behind if the list is empty :) there is no condition for that or handling this as returning specified sealed class object because of handling some specific situation in viewmodel or something :D so in this case i would really drop those ones :D event if it is some logic during the mapping, the mapper test could resolve those cases at all :D

Reply:
Done 😄